File: python3/1415.py
class Solution:
    def getHappyString(self, n: int, k: int) -> str:
        self.k = k

        def find(res, i, n):
            if i == n:
                self.k -= 1
                return self.k == 0
            for c in ["a", "b", "c"]:
                if res[i - 1] != c:
                    res[i] = c
                    if find(res, i + 1, n):
                        return True
            return False

        res = ["a"] * n
        for c in ["a", "b", "c"]:
            res[0] = c
            if find(res, 1, n):
                return "".join(res)
        return ""

File: java/1415.java
class Solution {
    int k;

    public String getHappyString(int n, int k) {
        char res[] = new char[n];
        this.k = k;
        for (char c = 'a'; c < 'd'; c++) {
            res[0] = c;
            if (find(res, 1, n))
                return new String(res);
        }
        return "";
    }

    private boolean find(char res[], int i, int n) {
        if (i == n) {
            k--;
            return k == 0;
        }
        for (char c = 'a'; c < 'd'; c++)
            if (res[i - 1] != c) {
                res[i] = c;
                if (find(res, i + 1, n))
                    return true;
            }
        return false;
    }
}
